src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -256,6 +259,20 @@ def run(self):
     fork_context = self._graph_helper.scheduler_session.with_fork_context if self._graph_helper else None
     self.daemonize(write_pid=False, fork_context=fork_context)
 
+  def pre_fork(self):
+    # Mark all services pausing (to allow them to concurrently pause), and then wait for them
+    # to have paused.
+    # NB: PailgunServer ensures that the entire run occurs under the lifecycle_lock.
+    for service in self._services.services:
```

Comment:
I was trying to reconcile whether the two steps are necessary since `mark_pausing` raises if its not in a `Running` state anyway. then realized the first step is merely a marker and the second is the one pausing.